src/main/java/pl/connectis/Spotify_API_CLI/authorization/Token.java

```diff
@@ -0,0 +1,100 @@
+package pl.connectis.Spotify_API_CLI.authorization;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.*;
+
+@Getter
+@Setter
+@ToString
+@EqualsAndHashCode
+@JsonIgnoreProperties(ignoreUnknown = true)
+@Slf4j
+public class Token implements Serializable {
+
+    private final Long generation_time = System.currentTimeMillis();
+    private final File codeFile = new File("token");
+    private String access_token;
+    private String token_type;
+    private String scope;
+    private Integer expires_in; //Seconds
+    private Long expirationTimeMillis;
+    private String refresh_token;
+
+    public Token() {
+    }
+
+    public void setExpirationTimeMillis() {
+        this.expirationTimeMillis = this.generation_time - 30000L + this.expires_in * 1000L; // 30 seconds overhead
+    }
+
+    public Token saveToken() {
+        Token tokenToSave = this;
+        setExpirationTimeMillis();
+
+        try {
+            FileOutputStream fileOut = new FileOutputStream(codeFile);
+            ObjectOutputStream out = new ObjectOutputStream(fileOut);
+            out.writeObject(tokenToSave);
+            out.close();
+            fileOut.close();
+        } catch (IOException ex) {
+            log.error("Saving file error " + ex.getMessage() + '\n' + "Path: " + codeFile.getPath());
+        }
+        return tokenToSave;
+    }
+
+    public Token readToken() {
```

Comment:
pakiet NIO javy dostarcza dużo łatwiejszych metod na czytanie plików - przerób to na NIO

Reply:
Ok, poczytam o tym.